mobly/controllers/android_device.py

```diff
@@ -207,8 +207,15 @@ def get_instances_with_configs(configs):
             raise Error(
                 "Required value 'serial' is missing in AndroidDevice config %s."
                 % c)
-        ad = AndroidDevice(serial)
-        ad.load_config(c)
+        is_required = c.pop("required", True)
+        try:
+            ad = AndroidDevice(serial)
+            ad.load_config(c)
+        except:
+            if not is_required:
```

Comment:
This part only covers adb responsiveness.
As for the services, I guess we should also skip errors for optional devices?
But then it bares the question of which services are essential and which are not.

If the user sets sl4a required to True and device's required to False, what should we do when the device is discoverable in adb yet sl4a fails to start?